components/api.rst

```diff
@@ -75,20 +75,35 @@ Configuration variables:
 
 .. _api-actions:
 
-.. note::
-
-    Before a newly added ESPHome device can interact with the Home Assistant API it needs to be allowed to communicate
-    with it. This setting can be found in the ESPHome integration (NOT in the Add-On) by clicking "CONFIGURE" for
-    that device and enabling the "Allow device to make service calls" option.
-
 Actions
 -------
 
+Before using any of the actions below, you'll need to tell Home Assistant to allow your device to
+make service calls.
+
+Open the ESPHome integration page on your Home Assistant instance:
+
+.. raw:: html
+
+    <a href="https://my.home-assistant.io/redirect/integration/?domain=esphome" target="_blank" rel="noreferrer noopener"><img src="https://my.home-assistant.io/badges/integration.svg" alt="Open your Home Assistant instance and show an integration." /></a>
+
+Then:
+
+#. Fnd your device in the device list
+#. Click the "configure" button next to it
+#. Check the "Allow the device to make Home Assistant service calls" box
+#. Then click "submit".
+
 .. _api-homeassistant_event_action:
 
 ``homeassistant.event`` Action
 ******************************
 
+.. note::
+
+    Be sure to :ref:`follow the instructions above <api-actions>` to tell Home Assistant to allow
+    your device to make service calls.
+
```

Comment:
I would think this instance of the note could be removed, since the paragraph it's referring to is immediately above, and fairly obvious now that the HA button is there.

Reply:
The problem is when you arrive at that section by clicking on it in the sidebar. You get taken to something like this: https://deploy-preview-3631--esphome.netlify.app/components/api.html?highlight=native+api#homeassistant-event-action

That section is completely out of your scroll view at that point and you may not think to scroll up and see what you're missing. That's exactly what led to my making this change in the first place: I clicked on a sidebar link, saw the instructions I thought I needed to follow, then spent the next hour trying to figure out why it wasn't working before going back and reading the entire page that I had previously thought was mostly about other native API stuff that I didn't need to know.

So I'll respectfully argue for keeping that one in, even if it is otherwise a bit superfluous.

Reply:
Fair enough - not an ideal solution, but I can't think of a better one right now.